test/VeTokenDistributor/VeTokenDistributor.spec.ts

```diff
@@ -0,0 +1,526 @@
+        })
+    })
+
+    describe("toggle checkpoint", async () => {
+        it("force error when called by non-admin", async () => {
+            await expect(veTokenDistributor.connect(alice).toggle_allow_checkpoint_token()).to.be.reverted
+        })
+
+        it("toggle checkpoint after several weeks", async () => {
+            const currentWeek = getWeekTimestamp(await getLatestTimestamp(), true)
+
+            // alice lock 100 PERP for 3 weeks
+            await vePERP.connect(alice).create_lock(parseEther("100"), currentWeek + 3 * WEEK)
+
+            await waffle.provider.send("evm_setNextBlockTimestamp", [currentWeek + 2 * WEEK + 2 * DAY])
+
+            await veTokenDistributor.connect(admin).toggle_allow_checkpoint_token()
+            // distribute 1000 PERP
+            await testPERP.mint(admin.address, parseEther("1000"))
+            await veTokenDistributor.connect(admin).burn(testPERP.address)
+
+            // week1: 1000/(7+7+2) * 7 = 437.5
+            // week2: 1000/(7+7+2) * 7 = 437.5
+            // week3: 1000/(7+7+2) * 2 = 125
+            expect(await veTokenDistributor.tokens_per_week(currentWeek)).to.be.eq(parseEther("437.499367043739809404"))
+            expect(await veTokenDistributor.tokens_per_week(currentWeek + WEEK)).to.be.eq(
+                parseEther("437.499367043739809404"),
+            )
+            expect(await veTokenDistributor.tokens_per_week(currentWeek + 2 * WEEK)).to.be.eq(
+                parseEther("125.001265912520381191"),
+            )
+
+            await waffle.provider.send("evm_setNextBlockTimestamp", [currentWeek + 3 * WEEK])
+
+            // alice withdraw veToken first, then create a new lock again
+            await vePERP.connect(alice).withdraw()
+            await vePERP.connect(alice).create_lock(parseEther("100"), currentWeek + 4 * WEEK)
+
+            // alice should be able to claim week2 & week3 fees, but not week1 because her lock was created during week1
+            // 437.5 + 125 = 562.5
+            const aliceRewards = parseEther("562.500632956260190595")
+            const aliceLockEnd = await vePERP.locked__end(alice.address)
+            const [aliceLockedAmountBefore] = await vePERP.locked(alice.address)
+            const latestTimestamp = await getLatestTimestamp()
+
+            await expect(veTokenDistributor.connect(alice)["claim()"]())
+                .to.emit(veTokenDistributor, "Claimed")
+                .withArgs(alice.address, aliceRewards, 1, 3)
+                .to.emit(vePERP, "Deposit")
+                .withArgs(alice.address, aliceRewards, aliceLockEnd, 0, latestTimestamp + 1)
+
+            const [aliceLockedAmountAfter] = await vePERP.locked(alice.address)
+
+            expect(aliceLockedAmountAfter.sub(aliceLockedAmountBefore)).to.be.eq(aliceRewards)
+        })
+    })
+
+    describe("recover balance", async () => {
+        beforeEach(async () => {
+            await testUSDC.mint(veTokenDistributor.address, parseUnits("1000", 6))
```

Comment:
Related to the `testUSDC` comment above. Please update the decimals here if necessary. Right now it is not consistent because the `testUSDC` in this file is 18 decimals

Reply:
fixed